### PR TITLE
Pull Request for TSE Review: Responsive Design (The Next Web) 

### PR DESCRIPTION
**Pull request by Calvinoea**
Hello,

I am making a request for TSE review of my work for the responsive design project which required a responsive clone to be made of [The Next Web](https://thenextweb.com/). 

The clone has been made using HTML and CSS. Media queries have been used to make the site responsive. [HTML validator](https://validator.w3.org/nu/#file) and stickler have been used to check for errors in the code. Please find below a link to the live demo:

[The Next Web](https://kind-bhaskara-e0bd5c.netlify.com/) 

I would like to give thanks to my former partner @mupa1 who started the project before she had to leave the programme. I was given the go-ahead by Microverse to complete the project on my own after her departure. 

Thank you.

### CODE REVIEW

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes

Hi @Calvinoea and @Mupa1, your project looks great 👍 good job 🙂 . However, we need to make a few changes to get the project approved. Find comments below:
- ```In medium``` and ```Small screen, ``` There is a horizontal scroll. Kindly fix it to make the project more responsive
- Kindly remove the commented code. ```Line 29``` of ```index.html```
- The 3 articles need some fixing. Kindly increase the width of the leftmost article. 
res4
Currently, this is what we have
res3
```Original version```
res2
- Currently, this is how the 3 articles section looks like on ```Medium screen```
res6
```Original version``` This is how it should look. With the TNW logo disappearing.
res5
- ```In the latest news section``` the articles are arranged in 3 articles per row. Currently, we do not have it. 
res7
```Original version``` for reference
res8
```In smaller version``` of ```latest news``` currently is not looking good. It aligns 1 article per row. Each article having a cover image on the left and title on the right. This is how it looks currently
res9
```Original version``` for reference
res10
- ```Latest funding section``` let us fix the header of latest funding on ```medium screens```. The header is overlapping with some texts
res11
- Let us fix the latest funding section on small screen. Let us increase the width of the article and the positions of the text. Check images for reference below:
**Currently**
res12
**Original**
res13
- Currently, in the ```medium screen``` of the apps, gear, etc. We need some fixing there. This is how it currently looks like
res14
**Original**
res15
- ```The latest deals``` align 3 deals per row. Currently, it looks a bit messy
res16
```Original version```
res17
- The latest deals look messy on small screens.
res18
```Original```
res19
Kindly fix the footer part of the project on a small screen and check the position of the TNW logo on medium screen
Currently
foo1
Original
foo

Make these changes and submit code review 👍 
Happy coding 💻 
Daniel Larbi Addo

**Comment by addod19** [comment]
Kindly fix the warning returned by w3validator

**Comment by Calvinoea** [comment]
> ## Status: Require changes
> Hi @Calvinoea and @Mupa1, your project looks great 👍 good job 🙂 . However, we need to make a few changes to get the project approved. Find comments below:
> 
> * `In medium` and `Small screen, ` There is a horizontal scroll. Kindly fix it to make the project more responsive
> * Kindly remove the commented code. `Line 29` of `index.html`
> * The 3 articles need some fixing. Kindly increase the width of the leftmost article.
>   res4
>   Currently, this is what we have
>   res3
>   `Original version`
>   res2
> * Currently, this is how the 3 articles section looks like on `Medium screen`
>   res6
>   `Original version` This is how it should look. With the TNW logo disappearing.
>   res5
> * `In the latest news section` the articles are arranged in 3 articles per row. Currently, we do not have it.
>   res7
>   `Original version` for reference
>   res8
>   `In smaller version` of `latest news` currently is not looking good. It aligns 1 article per row. Each article having a cover image on the left and title on the right. This is how it looks currently
>   res9
>   `Original version` for reference
>   res10
> * `Latest funding section` let us fix the header of latest funding on `medium screens`. The header is overlapping with some texts
>   res11
> * Let us fix the latest funding section on small screen. Let us increase the width of the article and the positions of the text. Check images for reference below:
>   **Currently**
>   res12
>   **Original**
>   res13
> * Currently, in the `medium screen` of the apps, gear, etc. We need some fixing there. This is how it currently looks like
>   res14
>   **Original**
>   res15
> * `The latest deals` align 3 deals per row. Currently, it looks a bit messy
>   res16
>   `Original version`
>   res17
> * The latest deals look messy on small screens.
>   res18
>   `Original`
>   res19
>   Kindly fix the footer part of the project on a small screen and check the position of the TNW logo on medium screen
>   Currently
>   foo1
>   Original
>   foo
> 
> Make these changes and submit code review 👍
> Happy coding 💻
> Daniel Larbi Addo

Hi Daniel. Thank you for your review. The changes will be made as discussed.

**Comment by uryelah** [comment]
### Code under review by Sarah Chamorro (TSE)

**Comment by uryelah** [comment]
Hello **Giancarlo Dumani Stradtmann** & **Alvaro Paz Rosero**, I'm Sarah, your reviewer for this PR . 😃 👋 

First of all, you did a great fixing the **Latest news** section on small screens.🕺 

--- 
**Great job for:**

- Use of gradients over images
- Use of semantic HTML
- Attention to text color and font
- Use of icons
-------

###  Requesting changes 🔧 🏗 

You've missed some of the other fixes required by the previous review though:

- [ ] **In medium and Small screen, There is a horizontal scroll. Kindly fix it to make the project more responsive** (TIP: check [similar issues on stackoverflow](https://stackoverflow.com/questions/25206243/debugging-css-page-overflow) and the `overflow` propriety)

- [ ] **Don't keep commented code in your files. Please remove the commented code at the style.css file** (line 14 - 41) 
You can refer to previous commits or another branch if you want to save the commented code for reference.

- [ ] **The TWN logo from the first image disappears in the medium size screens (around 100px width)**
Your version:
Screenshot_2020-02-28 The Next Web
Original page:
Screenshot_2020-02-28 TNW - International technology news, business and culture

- [ ] **Good try with the 3 articles but now they left ones are too small. Compare your version and the original one.**
Screenshot_2020-02-28 The Next Web(1)
Screenshot_2020-02-28 TNW - International technology news, business and culture(1)

- [ ] **Nice fix for the  **Latest funding** section for medium screens, but now the text is not properly vertically aligned in small screens (< 700px).**

Screenshot_2020-02-28 The Next Web(2)

- [ ] **There is some unwanted between *Launch* and *Distract* on big screen sizes. Please fix it so they are aligned with the section above.**

Screenshot_2020-02-28 The Next Web(7)


- [ ] **Currently, in the medium screen of the apps, gear, etc. (Please refer to the previous review)**

Screenshot_2020-02-28 The Next Web(3)

- [ ] **The latest deals align with 3 deals per row on medium screens and 1 per row on small screens. (Please refer to the last code review)**

Your version:
Screenshot_2020-02-28 The Next Web(4)

Screenshot_2020-02-28 The Next Web(5)

Original site:
Screenshot_2020-02-28 TNW - International technology news, business and culture(3)
Screenshot_2020-02-28 TNW - International technology news, business and culture(2)

- [ ] **On scroll some of your page text is overlapping with the fixed navbar. Please fix it's  `z-index`.**
Screenshot_2020-02-28 The Next Web(6)

---

This project is one of the most demanding ones in the HTML path, in my personal opinion, so it's ok to take a bit more time with it. Try to rethink how you are organizing some of your HTML as it can help with positioning and responsibility.

**Please kindly fix the REQUIRED above issues and send another code review request.**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by Calvinoea** [comment]
> Hello **Giancarlo Dumani Stradtmann** & **Alvaro Paz Rosero**, I'm Sarah, your reviewer for this PR . 😃 👋
> 
> First of all, you did a great fixing the **Latest news** section on small screens.🕺
> 
> **Great job for:**
> 
> * Use of gradients over images
> * Use of semantic HTML
> * Attention to text color and font
> * Use of icons
> 
> ### Requesting changes 🔧 🏗
> You've missed some of the other fixes required by the previous review though:
> 
> * [ ]  In medium and Small screen, There is a horizontal scroll. Kindly fix it to make the project more responsive (TIP: check [similar issues on stackoverflow](https://stackoverflow.com/questions/25206243/debugging-css-page-overflow) and the `overflow` propriety)
> * [ ]  Don't keep commented code in your files. Please remove the commented code at the style.css file (line 14 - 41)
> * [ ]  The TWN logo from the first image disappears in the medium size screens (around 100px width)
>   Your version:
>   Screenshot_2020-02-28 The Next Web
>   Original page:
>   Screenshot_2020-02-28 TNW - International technology news, business and culture
> * [ ]  Good try with the 3 articles but now they left ones are too small. Compare your version and the original one.
>   Screenshot_2020-02-28 The Next Web(1)
>   Screenshot_2020-02-28 TNW - International technology news, business and culture(1)
> * [ ]  Nice fix for the  **Latest funding** section for medium screens, but now the text is not properly vertically aligned in small screens (< 700px).
> 
> Screenshot_2020-02-28 The Next Web(2)
> 
> * [ ]  There is some unwanted between _Launch_ and _Distract_ on big screen sizes. Please fix it so they are aligned with the section above.
> 
> Screenshot_2020-02-28 The Next Web(7)
> 
> * [ ]  Currently, in the medium screen of the apps, gear, etc. (Please refer to the previous review)
> 
> Screenshot_2020-02-28 The Next Web(3)
> 
> * [ ]  The latest deals align with 3 deals per row on medium screens and 1 per row on small screens. (Please refer to the last code review)
>   Your version:
>   Screenshot_2020-02-28 The Next Web(4)
> 
> Screenshot_2020-02-28 The Next Web(5)
> 
> Original site:
> Screenshot_2020-02-28 TNW - International technology news, business and culture(3)
> Screenshot_2020-02-28 TNW - International technology news, business and culture(2)
> 
> * [ ]  On scroll some of your page text is overlapping with the fixed navbar. Please fix it's  `z-index`.
>   Screenshot_2020-02-28 The Next Web(6)
> 
> This project is one of the most demanding ones in the HTML path, in my personal opinion, so it's ok to take a bit more time with it. Try to rethink how you are organizing some of your HTML as it can help with positioning and responsibility.
> 
> **Please kindly fix the REQUIRED above issues and send another code review request.**
> 
> Cheers and Happy coding!👏👏👏
> 
> Code review by **Sarah Uryelah Silva Chamorro**
> 
> (Feel free to contact me on slack by the handle of **sarah** if you have any additional question)

Thank you for your feedback @uryelah . Changes will be made following your feedback.

File: styles.css [hunk 1]
```diff
@@ -0,0 +1,2923 @@
...
+  }
+
+  .caption1 {
+    background: url(https://cdn0.tnwcdn.com/wp-content/blogs.dir/1/files/2020/02/1-796x417.jpg) no-repeat;
+    background-size: cover;
+    grid-column-start: 1;
+    grid-row-start: 1;
+    grid-row-end: 5;
+
+    /* min-width: 893px; */
```
Comment (addod19) [inline styles.css]: Kindly delete all commented code as pointed out by the previous reviewers.

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes  🔴

Hi @Calvinoea, 👍 nice job fixing some of the changes requested by the previous reviewers. However, we still have some fixing to do. 🙂 Find comments below:
- Now the leftmost part of the row now has a width more than the 2/3 of the recommended width.  Consider using ```%```. For instance, the leftmost portion should be ```60%``` and rightmost ```30%``` that makes the leftmost 2/3 and rightmost 1/3 as recommended and the rest should be margin or possible play around with it. Kindly increase the size of the TNW logo. Currently, this is how it looks
out
```Original version```
out1
- Kindly position, the articles properly on a small screen device. Currently, this is how it looks. 
The first article is out of position.
out2
- Consider the width of the nav on a small screen device. Currently, this is how it looks
out3
```Original version```
out4
- The latest funding section looks a bit messy. Let us check it out and fix it. Currently, this is how it looks. Some text is missing and articles overlap on each other.
out5
- Let us fix the latest deals section. Currently, this is how it looks, let us try to make it look like the original. Let us compare the 2 pictures below and see where it needs fixing.
```Our version```
out6
```Original version```
out7
- Let us fix the position of stackpath logo at the footer. Currently, this is how it looks on small screen
out8
```Original version```
out9
- In medium screen size, currently, this is how the footer looks
ou
```Original version```
ou1
- In the latest deals on a medium screen, let us fix the position of the articles and arrange them like the original. Currently, this is how it looks
ou2
```Original version```
ou3
- Some articles overlap with others. Check image for reference
ou4
- Let us fix the positioning of the latest funding and how they are positioned
ou6
```Original version```
ou5

Kindly make these changes and submit another code review 👍 
Good luck with your next submission ✋ 
_Happy coding_ 💻 
Daniel Larbi Addo

**Comment by Calvinoea** [comment]
Thank you for your feedback @addod19 . I am working on the changes.

**Review by nikasulo — CHANGES_REQUESTED** [review]
# Under Review By Tolulope Olusakin(TSE)
Please check on this in a few minutes for some feedback :ok_hand:

**Comment by nikasulo** [comment]
# Great work @Calvinoea and @Mupa1 :clap: 
You have done nicely building this, however, I have noticed some serious issues with the responsiveness of your project and this being one of the most paramount requirements, I will be bringing it to your attention with a few screenshots I made. Please consider my comments below.

#### Responsiveness
One of the first issues I noticed is with the navigation bar, you have content overlapping it when you scroll, especially on mobile and tablet screen sizes(on desktop to, just a little less). There are also issues with section and element sizes, for instance, the first section on the page in both mobile and tablet screens have a number of visible bugs. Please consider the following screenshots, compare your project with the original on various screen sizes with the aid of the developer tools in the browser, make the necessary changes and request another review. 

screenshot
screenshot
screenshot
screenshot

I wish you the best and see you at the top!

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

**Comment by Calvinoea** [comment]
> # Great work @Calvinoea and @Mupa1 👏
> You have done nicely building this, however, I have noticed some serious issues with the responsiveness of your project and this being one of the most paramount requirements, I will be bringing it to your attention with a few screenshots I made. Please consider my comments below.
> 
> #### Responsiveness
> One of the first issues I noticed is with the navigation bar, you have content overlapping it when you scroll, especially on mobile and tablet screen sizes(on desktop to, just a little less). There are also issues with section and element sizes, for instance, the first section on the page in both mobile and tablet screens have a number of visible bugs. Please consider the following screenshots, compare your project with the original on various screen sizes with the aid of the developer tools in the browser, make the necessary changes and request another review.
> 
> screenshot
> screenshot
> screenshot
> screenshot
> 
> I wish you the best and see you at the top!
> 
> ##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

Thank you for your review @Oluwadamilareolusakin . I will make the necessary changes.

**Review by nikasulo — CHANGES_REQUESTED** [review]
# Hey team! :wave: 
Great work attempting to make changes to your project, I noticed a few things are different, however, many of the same problems I described and presented to you in the screenshots persist. Do you mind sending me a group message on Slack to discuss before your next review?

##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

**Comment by Calvinoea** [comment]
> # Hey team! 👋
> Great work attempting to make changes to your project, I noticed a few things are different, however, many of the same problems I described and presented to you in the screenshots persist. Do you mind sending me a group message on Slack to discuss before your next review?
> 
> ##### Think I missed something? Need some help with this or any of your projects? Please feel free to reach out to me on [Twitter](https://twitter.com/oluwadamilareo_) or [Instagram](https://instagram.com/oluwadamilare_olusakin), even if you just want to chat and I will respond to you as soon as I can.

Hello. I have sent you a message on Slack.

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team!
### Good job so far and also on implementing changes from previous reviews. However, you still have some changes to work on before your project could be approved.
- Validate your html code and solve the errors, [use this website](https://validator.w3.org/#validate_by_upload)
validator
- On mobile devices, some layout are off, make these changes
mobile 2
mobile 3
mobile
### [REQUIRED]
- Your website is not responsive enough for both mobile and tablets, use the bootstrap grid to achieve this
- Avoid using br tag in your html code, rather you can achieve this via box model
br
- Use html5 semantics tags like nav, section, article, main, aside where necessary, this will save you a lot of time.
### Make these changes and resubmit for another code review. Good luck.
- For assistance, reach out to me on slack @iamraymond
- Happy coding!

**Comment by Calvinoea** [comment]
> Hello Team!
> 
> ### Good job so far and also on implementing changes from previous reviews. However, you still have some changes to work on before your project could be approved.
> * Validate your html code and solve the errors, [use this website](https://validator.w3.org/#validate_by_upload)
>   validator
> * On mobile devices, some layout are off, make these changes
>   mobile 2
>   mobile 3
>   mobile
> 
> ### [REQUIRED]
> * Your website is not responsive enough for both mobile and tablets, use the bootstrap grid to achieve this
> * Avoid using br tag in your html code, rather you can achieve this via box model
>   br
> * Use html5 semantics tags like nav, section, article, main, aside where necessary, this will save you a lot of time.
> 
> ### Make these changes and resubmit for another code review. Good luck.
> * For assistance, reach out to me on slack @IAmRaymond
> * Happy coding!

Thank you, @rOluochKe . I am making the changes.

**Comment by uryelah** [comment]
Hello **Calvin Ebun-Amu** & **Mupa Nzaphila**, I'm Sarah, your reviewer for this PR 👩‍🚀  

Good job fixing some of the bugs from the past reviews, but I noticed that there still a lot of the layout is breaking, and some new bugs also appeared, though the project is way better since the first time I reviewed it.

If you need more assistance please reach out via[https://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-]( https://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-)

This is one of the most complex projects of the HTML path, so it's ok if you take some time to rework the markup so that it's easier to style and align.

--------

### Required changes 🛠 

- [ ] Please fix the articles of the site that are overlapping on medium to small screens:
Screenshot_2020-03-09 The Next Web(5)
Screenshot_2020-03-09 The Next Web(1)
Screenshot_2020-03-09 The Next Web
Screenshot_2020-03-09 The Next Web(6)

- [ ] The text of the three main articles should at the bottom of the image, with the gradient behind it.
Screenshot_2020-03-09 The Next Web(2)


- [ ] Please fix the text and nav area for small screens:
Screenshot_2020-03-09 The Next Web(3)




---------


**Please fix the required changes 👷  before asking for a new code review**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by Calvinoea** [comment]
> Hello **Calvin Ebun-Amu** & **Mupa Nzaphila**, I'm Sarah, your reviewer for this PR 👩‍🚀
> 
> Good job fixing some of the bugs from the past reviews, but I noticed that there still a lot of the layout is breaking, and some new bugs also appeared, though the project is way better since the first time I reviewed it.
> 
> If you need more assistance please reach out viahttps://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-
> 
> This is one of the most complex projects of the HTML path, so it's ok if you take some time to rework the markup so that it's easier to style and align.
> 
> ### Required changes 🛠
> * [ ]  Please fix the articles of the site that are overlapping on medium to small screens:
>   Screenshot_2020-03-09 The Next Web(5)
>   Screenshot_2020-03-09 The Next Web(1)
>   Screenshot_2020-03-09 The Next Web
>   Screenshot_2020-03-09 The Next Web(6)
> * [ ]  The text of the three main articles should at the bottom of the image, with the gradient behind it.
>   Screenshot_2020-03-09 The Next Web(2)
> * [ ]  Please fix the text and nav area for small screens:
>   Screenshot_2020-03-09 The Next Web(3)
> 
> **Please fix the required changes 👷 before asking for a new code review**
> 
> Cheers and Happy coding!👏👏👏
> 
> Code review by **Sarah Uryelah Silva Chamorro**
> 
> (Feel free to contact me on slack by the handle of **sarah** if you have any additional question)

Hey @uryelah . Thank you for the points of correction. I am currently working on making the changes.

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @Calvinoea  and @Mupa1 
### You have really tried and for also trying to make the changes from previous reviews. :clap: However, there are a few issues you need to implement before your project could be approved.
- Validate your html code and resolve the issues, [use this tool](https://validator.w3.org/#validate_by_upload)
not responsive 2
validator
not responsive 3
not responsive
- You need to put more work responsive design to achieve the desired layout for both phones and medium screens devices
not responsive 1
### Make these changes and resubmit for another code review, good luck.
- You can also reach out to any TSE to help you if you are facing challenges, it is good to ask for help.
- You can contact me on slack @iamraymond
- Happy coding

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team!
### Still not looking good, do you mind seeking help from either your mentors or any TSE
not looking good
res 2
res 3

### You can contact me as well on slack @iamraymond

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @Calvinoea and @Mupa1 
### Good job :clap: I see improvements but still not yet there, a few issues to work on.
- Concentrate mostly on the responsive design sizes medium: min-width around 768px
large: min-width 1024px
not good 2
not good
### Implement the changes and resubmit for another code review. Good luck
- Happy coding.

**Comment by Calvinoea** [comment]
> Hello @Calvinoea and @Mupa1
> 
> ### Good job 👏 I see improvements but still not yet there, a few issues to work on.
> * Concentrate mostly on the responsive design sizes medium: min-width around 768px
>   large: min-width 1024px
>   not good 2
>   not good
> 
> ### Implement the changes and resubmit for another code review. Good luck
> * Happy coding.

Thank you @rOluochKe . I am currently making the changes.

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
Project requires changes.

**Comment by pelzolga123** [comment]
## Hi Team!  
### You made some changes, but still a lot is messed up:
* Laptop version
image
Original:
image

* Tablet version
image

Original:
image

* Mobile version
image

Original:
image

### Please take a good look at original website. Your project should look like original. And only after your project looks like original, send another review request. 

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

**Comment by lekegitrepo** [comment]
### Hi @Calvinoea and @Mupa1 
### I would suggest that you should take some time to learn more about responsive web design before continue working on this project.

### All the changes required from the last review are not met.

### you can contact me on slack if you have any doubt or question.

### Reviewed by: Temitayo Adeleke

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Status: Require changes.

**Comment by uryelah** [comment]
## Calvin! You did it! 👏 👏 👏 

You've come a long way since the start of this project. Everything is responsive and neat.

There are some small issues here and there but the main structure is working and you should be really proud of yourself for achieving this!

## Your project is approved! 🎊 🥅 💛 
### 
**You may close this Pull Request and merge the changes**

-------

### Optional Improvement

- [ ] [OPTIONAL] The logo has some white margin already it and it's not scaling correctly. Consider fixing it and maybe using an `svg` instead of an image for it. [Read this cool article about it for more.](https://css-tricks.com/using-svg/)
Screenshot_2020-03-23 The Next Web


-------

**Keep being an awesome dev, good luck with your next project!**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**